### ISSUE TITLE
Integrating Sklearn into Syft

### ISSUE THREAD

**Issue by dnabanita7** [OpenMined/PySyft#5445]
## Feature Description

Since ``syft/lib/sklearn`` only makes use of ``Logistic Regression`` and ``Linear Regression``, I am planning on adding more modules, classes and methods such as 

**Base class sklearn.cluster**
- [ ] Birch
- [ ] DBSCAN
- [ ] KMeans
- [ ] MiniBatchKMeans

**Base class sklearn.datasets**
- [ ] fetch_california_housing
- [ ] load_boston
- [ ] load_breast_cancer
- [ ] load_diabetes
- [ ] load_digits
- [ ] load_files
- [ ] load_iris
- [ ] load_sample_image
- [ ] load_sample_images
- [ ] make_blobs

**Base class sklearn.decomposition**
- [ ] Fast ICA
- [ ] Incremental PCA
- [ ] Kernel PCA
- [ ] Latent Dirichlet Allocation
- [ ] PCA
- [ ] Sparse PCA
- [ ] Mini Batch Dictionary Learning
- [ ] sparse_encode

**Base class sklearn.discriminant_analysis**
- [ ] Linear Discriminant Analysis
- [ ] Quadratic discriminant Analysis

**Base class sklearn.Dummy**
- [ ] Dummy Classifier
- [ ] Dummy Regressor

**Base class sklearn.ensemble**
- [ ] AdaBoost Classifier
- [ ] AdaBoost Regressor
- [ ] Bagging Classifier
- [ ] Bagging Regressor
- [ ] Gradient Boosting Classifier
- [ ] Gradient Boosting Regressor
- [ ] Random Forest Regressor
- [ ] Random Forest Classifier

**Base class sklearn.gaussian_process**
- [ ] Gaussian Process regressor
- [ ] Gaussian Process classifier
- [ ] Gaussian_process kernels RBF

**Base class metrics**
- [ ] accuracy score
- [ ] auc
- [ ] classification report
- [ ] confusion matrix
- [ ] log loss
- [ ] f1 score
- [ ] multi label confusion matrix
- [ ] precision recall curve
- [ ] precision score
- [ ] recall score
- [ ] roc curve
- [ ] roc auc score
- [ ] top k accuracy score
- [ ] max error
- [ ] mean absolute error
- [ ] mean squared error

**Base class for sklearn.multioutput**
- [ ] MultiOutput Regressor
- [ ] Multioutput classifier

**Base class for sklearn.naive_bayes**
- [ ] GaussianNB
- [ ] BernoulliNB
- [ ] MultinomialNB

**Base class for sklearn.neighbors**
- [ ] KNeighbors Regressor
- [ ] KNeighbors Classifier

**Base class for sklearn.neural_networks**
- [ ] MLP classifier
- [ ] MLP regressor

**Base class for pre processing**
- [ ] scale
- [ ] min max scale
- [ ] normalize
- [ ] Standard Scaler
- [ ] One Hot Encoder
- [ ] Label Encoder

**Base class for sklearn.svm**
- [ ] Linear SVC
- [ ] Linear SVR
- [ ] SVR
- [ ] SVC

**Base class for sklearn.tree**
- [ ] Decision Tree Classifier
- [ ] Decision Tree Regressor

**Comment by Param-29** [comment]
I'd suggest changing label to GSoC to remove confusion.
